web/Services/MailService.cs

```diff
@@ -41,15 +48,34 @@ public MailService(IOptions<EmailConfigOptions> config)
             Text = msgBody
         };
         foreach (var msgToAddress in _config.Value.ToAddress)
+        {
             message.To.Add(new MailboxAddress(msgToName, msgToAddress));
+        }
 
         using var client = new SmtpClient();
-        client.Connect(_config.Value.Host, _config.Value.Port, true);
+        await client.ConnectAsync(_config.Value.Host, _config.Value.Port, true);
 
         // Note: only needed if the SMTP server requires authentication
-        client.Authenticate(_config.Value.UserName, _config.Value.Password);
+        await client.AuthenticateAsync(_config.Value.UserName, _config.Value.Password);
+
+        await client.SendAsync(message);
+        await client.DisconnectAsync(true);
+    }
 
-        client.Send(message);
-        client.Disconnect(true);
+    public async Task TrySendMsg(
+        string msgBody,
+        string subject,
+        string msgFromName = "",
+        string msgToName = ""
+    )
+    {
+        try
+        {
+            await SendMsg(msgBody,subject);
+        }
+        catch (Exception)
+        {
+            // ignored
```

Comment:
We should log the exception

Reply:
We want to know what the exception was too, so that we can try and work out why it didn't send

Reply:
ah ok